packages/stylist-macros/src/tokentree/output/mod.rs

```diff
@@ -0,0 +1,44 @@
+use proc_macro2::TokenStream;
+
```

Comment:
The `Output*` types are there to mirror the structure in `stylist::ast`. Going through them 1-to-1 should give a nice readable way to follow changes to the ast and how each struct there is generated.

Reply:
I see. I guess we can keep the `Output*`, but I still kind of want to move parsing logic of selectors out of it.

I have also noticed a lot of following pattern:

```rust
        let ident_scopes = Ident::new("scopes", Span::mixed_site());
        let Self { contents } = self;

        quote! {
            {
                let #ident_scopes: ::std::vec::Vec::<::stylist::ast::ScopeContent> = ::std::vec![
                    #( #contents, )*
                ];
                ::stylist::ast::Sheet::from(#ident_scopes)
            }
        }
```

over

```rust
quote! { ::stylist::ast::Sheet::from(::std::vec![#( #contents, )*]) }
```

I thought `Span`s are used for when procedural macros are having an error to emit diagnostic information.
Is there other advantage of of casting a span on the generated token if it can never error here?

Reply:
Assigning to `#ident_scopes` here to make the type checker happy. An alternative would be

`quote! { ::stylist::ast::Sheet::from(::std::vec![#( ::stylist::ast::ScopeContent::from(#contents), )*]) }`

Which reminds me to put a `use std::convert::{Into, From};` into the quote.